packages/core-snapshots/src/workers/worker.ts

```diff
@@ -0,0 +1,159 @@
+import { Models, Utils } from "@arkecosystem/core-database";
+import { Container } from "@arkecosystem/core-kernel";
+import { Transactions as MagistrateTransactions } from "@arkecosystem/core-magistrate-crypto";
+import { Transactions } from "@arkecosystem/crypto";
+import { Connection, createConnection, getCustomRepository } from "typeorm";
+import { parentPort, workerData } from "worker_threads";
+
+import * as Codecs from "../codecs";
+import { StreamReader, StreamWriter } from "../codecs";
+import { Repository, Worker, WorkerAction } from "../contracts";
+import { Identifiers } from "../ioc";
+import * as Repositories from "../repositories";
+import * as Actions from "./actions";
+import { Application } from "./application";
+
+let app: Application;
+let action: WorkerAction;
+const _workerData: Worker.WorkerData = workerData;
+
+/* istanbul ignore next */
+const connect = async (options: any): Promise<Connection> => {
+    return createConnection({
+        ...options.connection,
+        namingStrategy: new Utils.SnakeNamingStrategy(),
+        entities: [Models.Block, Models.Transaction, Models.Round],
+    });
+};
+
+export const init = async () => {
+    Transactions.TransactionRegistry.registerTransactionType(MagistrateTransactions.BridgechainRegistrationTransaction);
+    Transactions.TransactionRegistry.registerTransactionType(MagistrateTransactions.BridgechainResignationTransaction);
+    Transactions.TransactionRegistry.registerTransactionType(MagistrateTransactions.BridgechainUpdateTransaction);
+    Transactions.TransactionRegistry.registerTransactionType(MagistrateTransactions.BusinessRegistrationTransaction);
+    Transactions.TransactionRegistry.registerTransactionType(MagistrateTransactions.BusinessResignationTransaction);
+    Transactions.TransactionRegistry.registerTransactionType(MagistrateTransactions.BusinessUpdateTransaction);
+
+    app = new Application(new Container.Container());
+
+    if (_workerData.connection) {
+        /* istanbul ignore next */
+        app.bind(Identifiers.SnapshotDatabaseConnection).toConstantValue(
+            await connect({ connection: _workerData.connection }),
+        );
+    }
+
+    /* istanbul ignore next */
+    app.bind<Repository>(Identifiers.SnapshotRepositoryFactory).toFactory<Repository>(
+        (context: Container.interfaces.Context) => (table: string) => {
+            if (table === "blocks") {
+                return getCustomRepository(Repositories.BlockRepository);
+            } else if (table === "transactions") {
+                return getCustomRepository(Repositories.TransactionRepository);
+            } else {
```

Comment:
Remove the `else` because if any of the previous statements are true we won't hit the `else` anyways and nothing happens after this `else`.